src/System.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -83,41 +83,58 @@ public AsymmetricSignatureProvider(AsymmetricSecurityKey key, string algorithm,
                 throw new ArgumentOutOfRangeException("key.KeySize", key.KeySize, string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10630, key.GetType(), SignatureProviderFactory.MinimumAsymmetricKeySizeInBitsForVerifying));
             }
 
+#if DNXCORE50
+            hash = GetHashAlgorithmName(algorithm);
+#else
             hash = GetHashAlgorithm(algorithm);
+#endif
             RsaSecurityKey rsaKey = key as RsaSecurityKey;
+
             if (rsaKey != null)
             {
+#if DNXCORE50
+                rsaCng = new RSACng();
```

Comment:
"Always" is a tough thing to say; because who knows when CNGNG will come out :)

But, yes, RSACng is always available for CoreFX <strong>on Windows</strong>.